### ISSUE TITLE
ignore blank strings in numeric filters

### ISSUE THREAD

**Issue by goravbhootra** [rcdilorenzo/filtrex#52]
Hi,

I have setup a filter as:
```
number :status, allow_decimal: false, allowed_values: 1..5
```

if no status is selected in filters, blank value is passed in params. I am filtering the blank values before sending the filter params to `Filtrex.parse_params` else it responds with an error. This pattern is so common, I was wondering if it would be good idea to include this as an option in the config - something like:
```
number :status, allow_decimal: false, allowed_values: 1..5, ignore_blanks: true
```

**Comment by rcdilorenzo** [comment]
When you say a "blank value" do you mean an empty string or `nil`? I like this proposal but want to make sure I understand for sure beforehand.

**Comment by goravbhootra** [comment]
html form sends empty string. I am applying `|> Enum.reject(fn {_, v} -> v == ""  end)` to filter out the values but it might be good idea to ignore empty string ```""``` and ```nil``` both.

**Comment by rcdilorenzo** [comment]
So, does `parse_params` actually throw an error if you pass an empty string or did you mean your application logic rejects it since it is then filtering using that blank parameter?

**Comment by goravbhootra** [comment]
```parse_params``` throws an error since ```""``` is not in the ```allowed_values```

**Comment by rcdilorenzo** [comment]
Oh, I see. Yes, I like this! What do you think about calling it `ignore_empty` versus `ignore_blanks`? I'm not sure myself and wonder how those strike you.

**Comment by goravbhootra** [comment]
```ignore_empty``` sounds more apt if you are planning to remove only empty strings whereas ```ignore_blanks```, to me, covers empty strings as well as ```nil```. Users will be able to relate to either of them, as long as documentation is clear about what it covers IMHO.

**Comment by rcdilorenzo** [comment]
I agree. Let's go with `ignore_blanks` then and have that match `""` and `nil`.
